docs/datamodel/index.rst

```diff
@@ -1,50 +1,116 @@
-and **computed** properties or links to define useful expressions
-(e.g. ``property email := .user_name ++ '@' ++ .provider_name``).
-
-**Expression Aliases** let you use existing types under new names to build
-on them without touching the original -- both in your schema or on the fly
-inside a query. You can also create your own **functions**, strongly typed
-along with everything else in EdgeDB. All this goes into the schema under
-one or more **modules** (namespaces).
+========
+Concepts
+========
 
+In this section you'll find an overview of EdgeDB, a relational
+database with strongly typed schema, starting from the database server
+all the way to the fundamental types, objects, and concepts for
+EdgeDB.
 
 .. toctree::
     :maxdepth: 3
     :hidden:
 
     overview
-    objects
+    objects/index
     scalars/index
     colltypes
     abstract
     functions
     links
     props
+    linkprops
     computables
     indexes
     constraints
     aliases
     annotations
-    modules
-    databases
     extensions
+
+
+Instances
+=========
+
+A running EdgeDB server is called an **instance**. Typically, you
+would create an instance for a :ref:`project
+<ref_cli_edgedb_project>`. You can :ref:`start
+<ref_cli_edgedb_instance_start>`, :ref:`stop
+<ref_cli_edgedb_instance_stop>`, and otherwise manage your instances
+using :ref:`ref_cli_edgedb_instance` commands.
+
+.. _ref_datamodel_databases:
+
+Databases
+=========
+
+An EdgeDB instance can have multiple databases in it. Every instance
+is created with an empty database called "edgedb". If you wish to add
+more, you can use the :eql:stmt:`CREATE DATABASE` EdgeQL command.
+Conversely, the :eql:stmt:`DROP DATABASE` command removes a database.
+
+The following command will get a list of all databases present in the
+instance:
+
+.. code-block:: edgeql
+
+    SELECT sys::Database.name;
+
+.. _ref_datamodel_modules:
+
+Modules
+=======
+
+Every database has a schema that fully describes it. Schema consists
+of logical units called  **modules**, which act as namespaces.
+Modules contain user-defined types, functions, etc. A module has a
+name that is unique inside a database. The same schema object name can
+be used in different modules without conflict.  For example, both
+``module1`` and ``module2`` can contain a ``User`` object type.
+
+Schema objects can be referred to by a fully-qualified name using the
+``<module>::<name>`` notation.
+
+Every EdgeDB schema contains the following standard modules:
+
+* ``std``: standard types, functions and other declarations
+* ``schema``: types describing the :ref:`introspection <ref_eql_introspection>`
+  schema
+* ``sys``: system-wide entities, such as user roles and
+  :ref:`databases <ref_datamodel_databases>`
+* ``cfg``: configuration and settings
+* ``stdgraphql``: GraphQL-related types
+* ``math``: algebraic and statistical functions
+* ``default``: the default module for user-defined types, functions, etc.
```

Comment:
Add a TODO here to link to the future `Standard Library` section.

Reply:
Added